### ISSUE TITLE
same interface for plugins and backends

### ISSUE THREAD

**Issue by markus2330** [ElektraInitiative/libelektra#973]
Currently, elektraPluginSet is called twice on success for resolver plugins, once for preparation, the second time for commit. Backends, however, are only called once.

This behaviour causes troubles when plugins are added at multiple positions. Ideally, the interface to a backend and to plugins should be identical. Thus I propose to refactor backends to be plugins which have a `elektraPluginCommit`.

Note that this proposal does not solve every problem that might occur when plugins are added at multiple positions, so we should also discuss an integer that tells in which position a plugin currently is called, similar to the hook interface #867.

**Comment by markus2330** [comment]
@vLesk is this already done?

**Comment by vLesk** [comment]
I'm currently working on the backend plugin, I'll try to get it done by the end of the coming week. Should I also implement the integer?

**Comment by markus2330** [comment]
> I'm currently working on the backend plugin, I'll try to get it done by the end of the coming week.

Very good. Please keep reporting or create a PR where you push commits from time to time.

> Should I also implement the integer?

Only if you need it. The more simple the interface between backend plugins and plugins is, the better.

**Comment by markus2330** [comment]
@vLesk is this issue closed by #2969?

**Comment by vLesk** [comment]
@markus2330 yes, it will be closed once #2969 is merged.
